pkg/daemon/daemon.go

```diff
@@ -1470,9 +1470,16 @@ func (dn *Daemon) removeUpdateInProgressTaint(ctx context.Context) error {
 	})
 }
 
-func (dn *Daemon) isLayeredNode() bool {
+func (dn *Daemon) isLayeredNode() *CoreOSDaemon {
 	_, ok := dn.node.Annotations[constants.DesiredImageConfigAnnotationKey]
-	return ok
+	if ok {
+		if dn.os.IsCoreOSVariant() {
+			return &CoreOSDaemon{dn}
+		} else {
```

Comment:
Should we degrade or something here?

Reply:
I don't think degrading is the right choice bc there isn't technically a problem with the cluster.

Reply:
Okay so go ahead and perform a normal update, even though the node is labeled layered?

Reply:
I agree with Colin below that a non-CoreOS node just shouldn't hit this via the Controller

Reply:
Oh okay yeah that'd be better. I'll look into fixing that on the controller side...